14-minimal/test/run

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+#
+# The 'run' performs a simple test that verifies that STI image.
+# The main focus here is to exercise the STI scripts.
+#
+# IMAGE_NAME specifies a name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+# EXPRESS_REVISION specifies which express.js branch or tag should be tested;
+# by default it uses the latest released version as reported by
+# `npm show express version`.
+
+THISDIR=$(dirname ${BASH_SOURCE[0]})
+
+TEST_LIST_APP="\
+test_run_app_application
+test_s2i_usage
+test_scl_usage
+test_connection
+test_docker_run_usage
+test_scl_variables_in_dockerfile
+test_npm_functionality
+test_nodemon_removed
+test_npm_cache_cleared
+test_npm_tmp_cleared
+kill_test_application
+test_dev_mode_true_development
+test_dev_mode_false_production
+"
+
+TEST_LIST_HW="\
+test_safe_logging
+test_run_hw_application
+"
+source "${THISDIR}/test-lib.sh"
+source "${THISDIR}/test-lib-nodejs.sh"
+
+test -n $IMAGE_NAME \
+  -a -n $VERSION
+readonly EXPRESS_REVISION="${EXPRESS_REVISION:-$(docker run --rm "${IMAGE_NAME}" -- npm show express version)}"
+
+test_dir="$(readlink -f $(dirname ${BASH_SOURCE[0]}))"
+image_dir="$(readlink -f ${test_dir}/..)"
+test_short_summary=''
+TESTSUITE_RESULT=0
+cid_file=$(mktemp -u --suffix=.cid)
+
+# Since we built the candidate image locally, we don't want S2I attempt to pull
+# it from Docker hub
+s2i_args="--pull-policy=never "
+
+# TODO: This should be part of the image metadata
+test_port=8080
+
+# Common git configuration
+readonly -A gitconfig=(
+  [user.name]="builder"
+  [user.email]="build@localhost"
+  [commit.gpgsign]="false"
+)
+
+
+if [ "$DEBUG" != "" ]; then
+  set -x
+fi
+
+# Build the application image twice to ensure the 'save-artifacts' and
+# 'restore-artifacts' scripts are working properly
```

Comment:
@pkubatrh Could I ask where does the `restore-artifacts` script reside? I can't find it.

Reply:
The script lives over here for the minimal image: https://github.com/sclorg/s2i-nodejs-container/pull/269/files#diff-e1328aa75681fcbf49d9fc0afa1335663cdd82b0f6d9979c1e63afa89802ceb7
But I do not think it is expected to be used in minimal and I am not sure how accurate that comment in the test suite is nowadays (it was added with rh-nodejs4!)
iirc we are testing the `save-artifacts` and `restore-artifacts` properly during incremental builds (https://github.com/sclorg/s2i-nodejs-container/pull/269/files#diff-327d7617e61c0674d3c54bcdf07af0e97e8f2c4b05213916558f5b76ced1b41fR326)

Reply:
I guess `restore-artifacts` is currently part of the assemble script: https://github.com/sclorg/s2i-nodejs-container/blob/master/14/s2i/bin/assemble#L30

Reply:
Considering this change will need to happen for all the nodejs images and the change does not have any immediate value for this PR I will make the change separately in a different PR.